src/commands/cmd_server.cc

```diff
@@ -823,16 +823,33 @@ class CommandScan : public CommandScanBase {
     if (args.size() % 2 != 0) {
       return {Status::RedisParseErr, errWrongNumOfArguments};
     }
-
-    ParseCursor(args[1]);
-    if (args.size() >= 4) {
-      ScanParameters parameters;
-      Status s = ParseScanParameters(args, &parameters);
-      FillScanAttributes(&parameters);
-      if (!s.IsOK()) {
-        return s;
+    CommandParser parser(args, 0);
+    while (parser.Good()) {
+      if (parser.EatEqICase("scan")) {
+        ParseCursor(GET_OR_RET(parser.TakeStr()));
```

Comment:
It's not a correct implementation. e.g. please consider if the input is `match ..`. `SCAN` itself cannot be parsed as an additional flag.
You should implement it like:
```
CommandParser parser(args, 1);
ParseCursor(GET_OR_RET(parser.TakeStr()));
while (..) ...
```